ThreeBotPackages/threefold/tfgrid_workloads/docs/reservation.md

```diff
@@ -0,0 +1,94 @@
+threebot (and possibly other threebots who will sign to trigger actions).
+
+## Reservation data
+
+The reservation data is composed of the individual workloads to be deployed, information
+about who can sign for the actual deployment and deletion of the workloads, and expiry
+times for the reservation. All fields in the data object are immutable after the
+reservation is created (i.e. the customer signs the data). Modifications afterwards
+will cause the customer signature to become invalid.
+
+## Reservation state
+
+The reservation state is updated throughout the lifetime of the reservation. It also
+contains the signatures needed to have the farmer threebot take action. In order for
+the farmer threebot to start provisioning the workloads, or delete the workloads,
+the `signing_request_provision` and `signing_request_delete`, respectively, need to be filled with
+valid signatures. A valid signature is a signature for the reservation data, with a
+private key owned by one of the threebots listed in the reservation data (in the
+`signatures_provision` and `signatures_delete` fields). These fields also define
+the minimum amount of signatures required. For example, a signature request for
+provisioning might list 3 threebot ids which can sign, but only specify a `quorum_min`
+of 2. As such, only 2 out of the 3 listed threebot ids would need to sign before the
+farmer threebot is allowed to deploy the reserved workloads.
+
+### signature validity
+
+A signature is created by signing a piece of data using a private key. Afterwards,
+the corresponding public key can be used to check if the signature is valid. A
+[signature field](#signature.signing) is valid if it meets the following conditions:
+
+- It contains at least the minimum amount of signatures required, as defined in
+the `quorum_min` field of the corresponding [signing request field](#signing.request), 1 if there is no such
+corresponding signing request.
+- All signatures are valid with a public key owned by a referenced threebot (referenced
+in the aforementioned accompanying [signing request field](#singing.request) or possibly other field).
+
+## models
+
+The following is an overview of the models available in the [models](./models) subdirectory,
+their fields, and what these fields are used for.
+
+### reservation
+
+The [reservation object](./models/reservation.toml) is the high level object for dealing with a reservation
+on the threefold grid. It is composed of the [reservation.data](#reservation.data) object, which holds
+all data for the workloads covered by this reservation (and is immutable after it is created),
+and additional info for the reservation state.
+
+- data: As explained, the data field holds the [reservation.data](#reservation.data) object, which contains all
+low level workloads to be provisioned. After the reservation is created, this field is
+immutable.
+- json: A representation of the `data` object, in json form. It is directly derived from
```

Comment:
what's the point of this field ? this is just a plain copy of the data ?

Reply:
yes, and as far as I understand, this will be the input for the signature algorithm (which I now also added in the description)